### DISCUSSION TITLE
suggest to add new finction for pyton Mtproto proxy script

### DISCUSSION THREAD

**Discussion by mahansalimi** [HirbodBehnam/MTProtoProxyCentOSInstall#42] [Ideas]
1.very perfect script.it will be better if could set quota by MB and KB by writing MB or KB after digit
2.add a function to reset a special user quota and a finction to show reached quota of user.

**Comment by HirbodBehnam** [comment]
Hello
Yes that's a good idea! I will probably work on it if I have some time...
The second one is also good. I will investigate how to implement it.

**Reply by mahansalimi** [reply]
also if after enter to program ,when show menu 1-8 items ,its good if write version of program or add a item 9 for about menu.

**Reply by HirbodBehnam** [reply]
Ok; I will add this one too!

**Comment by HirbodBehnam** [comment]
I added all of these in 5205628784495e20894a3e12bf84435b7f86808a!

**Reply by mahansalimi** [reply]
perfect  ,thanks for spend time to improve script.
a function is missed at this script.
admin cant check users used traffics and reset it.
i think in future if you have time ,add a menu to show users and their seted qouta and used qouta.

for example

`
User1    SetedQouta=2000MB          UsedTraffic=1MB           RemainTraffic=1000MB        Active       
User2    SetedQouta=10000MB        UsedTraffic=1000MB     RemainTraffic=0MB              DeActivate    

Enter user to rest used qouta:

`

its possible to add this function or there is restrict in functions?

**Reply by HirbodBehnam** [reply]
I'm sorry I saw your comment now. I'm not really sure if this is possible because the proxy itself prints its data on console in plain text. Might be possible to use regex to extract info.
Will investigate this...

**Comment by cline-pk** [comment]
you can add crontab to auto run mtproxy when ever server reboot.

`@reboot nohup python3 /opt/mtprotoproxy/mtprotoproxy.py &`

**Reply by HirbodBehnam** [reply]
Hello
The script actually starts the proxy on start up using `systemd`. It registers the proxy as a service called `mtprotoproxy` and then adds it to start up.
